README.md

```diff
@@ -12,7 +12,7 @@ Applications using the LaunchDarkly Go SDK will generally use the `ldcontext` su
 
 ## Supported Go versions
 
-This version of the project requires a Go version of 1.18 or higher.
+This version of the project requires a Go version of 1.24 or higher.
```

Comment:
LD policy states we do a rolling support of the two last versions, so we need to do 1.23+

```suggestion
This version of the project requires a Go version of 1.23 or higher.
```